### PR TITLE
feat(model-server): Add healthcheck to model-server image

### LINKED ISSUE #501: Add a health check to the model-server image
Currently we have to specify the health_check in each compose file manually. If the model-server image was making use of the [HEALTHCHECK](https://docs.docker.com/engine/reference/builder/#healthcheck) we could leave that responsibility to modelix.

### PR DESCRIPTION
**Pull request by drik98**
Closes #501

### CODE REVIEW

**Comment by drik98** [comment]
As not configured differently the defaults will be 

interval - DURATION (default: 30s)
timeout - DURATION (default: 30s)
start-period - DURATION (default: 0s)
retries - DURATION (default: 3)

which seems sensible

**Review by odzhychko — DISMISSED** [review]
HEALTHCHECK seems helpful.

Even if HEALTHCHECK does additional checks at start up, I cannot think of a case where this might break a realistic uses case of the image.

**Comment by odzhychko** [comment]
@drik98 Your commit is not signed.

> The base branch requires all commits to be signed. [Learn more about signing commits.](https://docs.github.com/articles/about-gpg/)

**Review by slisson — CHANGES_REQUESTED** [review]
There is a `/health` endpoint that should be used. See https://github.com/modelix/modelix.kubernetes/blob/1f62139d76482d0428a86c9fa167b7f38b441fec/helm/modelix/templates/common/model-deployment.yaml#L45-L58

**Comment by drik98** [comment]
> There is a `/health` endpoint that should be used. See https://github.com/modelix/modelix.kubernetes/blob/1f62139d76482d0428a86c9fa167b7f38b441fec/helm/modelix/templates/common/model-deployment.yaml#L45-L58

I meant to use that but forgot in the implementation. Good catch.

File: model-server/Dockerfile [hunk 1]
```diff
@@ -1,6 +1,7 @@
 FROM registry.access.redhat.com/ubi8/openjdk-11:1.18-2.1705602259
 WORKDIR /usr/modelix-model
 EXPOSE 28101
+HEALTHCHECK CMD curl --fail http://localhost:28101 || exit 1
```
Comment (languitar) [inline model-server/Dockerfile:4]: We have a dedicated health endpoint that can be used here. Cf. https://github.com/modelix/modelix.core/blob/aa16a542008905c49cc465a535a2557aead9369d/docs/global/modules/core/pages/howto/usage-model-server.adoc?plain=1#L70

```suggestion
HEALTHCHECK CMD curl --fail http://localhost:28101/health
```

What was the reason to add `|| exit 1`? Curl would fails with a non-zero exit code in any case.

**Review by languitar — CHANGES_REQUESTED** [review]

File: model-server/Dockerfile (hunk 1 above)
Comment (languitar) [reply model-server/Dockerfile:4]: Oops, now the remaining comments pointing to that loaded :see_no_evil:

Comment (drik98) [reply model-server/Dockerfile]: > What was the reason to add || exit 1? Curl would fails with a non-zero exit code in any case.

The reason is that curl may exit with any code https://timi.eu/docs/anatella/5_1_9_1_list_of_curl_error_codes.html but the health check should exit with either 0, 1 or 2. But it should not even be exited with 2 :D 


image

**Review by odzhychko — CHANGES_REQUESTED** [review]
Please change your three commits into one.

**Review by odzhychko — APPROVED** [review]

File: model-server/Dockerfile (hunk 1 above)
Comment (languitar) [reply model-server/Dockerfile]: Ah right, didn't know that Docker healthchecks are that picky.

**Review by languitar — APPROVED** [review]
Thanks for the PR. Looks fine to me.

**Review by slisson — APPROVED** [review]

**Pull request merged**